byterun/win32.c

```diff
@@ -57,10 +57,24 @@
 #endif
 
 /* Very old Microsoft headers don't include intptr_t */
-#if defined(_MSC_VER) && !defined(_UINTPTR_T_DEFINED)
+#ifdef _MSC_VER
+# ifndef _UINTPTR_T_DEFINED
+#ifdef _WIN64
```

Comment:
Just use `uintnat` and `intnat` defined in `caml/config.h`, these integer types are guaranteed to have the same size as pointer types:
```
typedef uintnat uintptr_t;
```